### ISSUE TITLE
p5js WEBGL textureWrap can't collaborate with video streaming 

### ISSUE THREAD

**Issue by 200275zhx** [processing/p5.js#5859]
### Topic

I recently created a video streaming sketch with p5js WEBGL. The content of the code is: create video capture, then render it on a cube as texture, and then share it with other clients using this library p5LiveMedia. 
When I finish, I try to view the video capture on two different tabs, but the video was not successfully rendered on Chrome. At first, I thought it was a problem with the browser, because it worked on safari, but error line is showing on neither browser's console.
Eventually, I found that it was because I used this line: texture mode (REPEAT); 
The problem is solved, but does anyone know why might this happen?

This is my project demo, you are welcome to experience it:
https://editor.p5js.org/hershey_zhang/sketches/lK4ZpM1hA

**Comment by davepagurek** [comment]
According to the docs, REPEAT only works on images/canvases where the side lengths are powers of two. Does it work if you create a canvas with power-of-two sides?

**Comment by 200275zhx** [comment]
No, my video is rendered as a texture on a box(2);, so it's probably not because of that.😂


> According to the docs, REPEAT only works on images/canvases where the side lengths are powers of two. Does it work if you create a canvas with power-of-two sides?

**Comment by davepagurek** [comment]
Sorry, when I said canvas, I mean the canvas for your video. Anything used as a texture with REPEAT or MIRROR is only supported if the texture has power-of-two sides

**Comment by 200275zhx** [comment]
😳Thank you for your clarification. In my original code, I used createCapture and didn't set a size for it. 
To verify your statement, I just tried to set the size of this video variable to (2, 2), but nothing changed. 

> Sorry, when I said canvas, I mean the canvas for your video. Anything used as a texture with REPEAT or MIRROR is only supported if the texture has power-of-two sides

**Comment by davepagurek** [comment]
Ah ok, it was worth a shot! Would you mind explaining how to edit the sketch you linked to test out the `textureMode(REPEAT)` mode? Just uncommenting that line didn't seem to make a difference, but maybe I also am looking at the wrong things.

**Comment by 200275zhx** [comment]
Yes, so I create Capture in sketch.js, 
<img width="802" alt="image" src="https://user-images.githubusercontent.com/42470637/201433555-ecd7b5e9-0a28-42d7-9351-b849e314d312.png">

And render it as texture no a box in players.js
<img width="693" alt="image" src="https://user-images.githubusercontent.com/42470637/201433398-9bc91ffa-fdd3-4de5-ba83-841f71a37975.png">

And I call it in draw function whenever a new user enter this room
<img width="726" alt="image" src="https://user-images.githubusercontent.com/42470637/201433772-1b8b50f8-8728-4697-9e65-896a2fe1d3cd.png">

In addition, uncommenting that line textureMode(REPEAT) will make a difference: 
When you open two tabs in chrome  and run this code at the same time, if textureMode(REPEAT) is uncommented, then only in one tab you can see the video rendered successfully. If textureMode(REPEAT) is commented, both tab renders video capture successfully.
